File: hacker_Rank/containsCommonItens.js
function containsCommonItem(arr1,arr2){
    const result = arr1.some(item => arr2.includes(item))
    console.log(result)
    return result
}

function containsCommonItem(arr1,arr2){
    let map = {}
    for(let i = 0;i<arr1.length;i++){
        if(!map[arr1[i]]){
            const item = arr1[i]
            map[item] = true;
        }
    }
    
    for(let i = 0;i<arr2.length;i++){
        if(map[arr2[i]]){
            return true
        }
    }
    console.log(false)
    return false
}

containsCommonItem(['a','b','c'],['x','a'])

File: leet_Code/containsCommonItens.py
# given 2 arrays, return true if they have common elements and false if not
# ['a','b','c'] ['x','y'] ['o','a','x']
# Learning hash Table
from typing import List


class Solution:
    def containsCommonItens(arr1: List[int], arr2: List[int]) -> bool:
        numMap = {}
        n1 = len(arr1)
        n2 = len(arr2)

        for i in range(n1):
            if (arr1[i] not in numMap):
                item = arr1[i]
                numMap[item] = True
        
        for i in range(n2):
            if(arr2[i] in numMap):
                print(True)
                return True
        print(False)
        return False


    containsCommonItens(['a','b','c'], ['x','c'])